token/program-2022/src/lib.rs

```diff
@@ -121,6 +121,14 @@ pub fn check_zk_token_proof_program_account(zk_token_proof_program_id: &Pubkey)
     Ok(())
 }
 
+/// Checks that the supplied program ID is correct for the spl record program
+pub fn check_spl_record_program_account(spl_record_program_id: &Pubkey) -> ProgramResult {
+    if spl_record_program_id != &spl_record::id() {
+        return Err(ProgramError::IncorrectProgramId);
+    }
+    Ok(())
+}
```

Comment:
Do we need to enforce this? If not, it might be simpler to remove this check and the dependency from token-2022 to the record program

Reply:
Yeah I thought about this as well. We don't necessarily need the proof to come from the record program, but I thought I would add it for safety. You are right. I think it will be simpler remove it and remove the dependency on the record program.